controls/V-72223.rb

```diff
@@ -1,77 +1,85 @@
-level if multiple application sessions are using a single operating system-level
-network connection. This does not mean that the operating system terminates all
-sessions or network access; it only ends the inactive session and releases the
-resources associated with that session.
+includes, for example, de-allocating associated TCP/IP address/port pairs at
+the operating system level and de-allocating networking assignments at the
+application level if multiple application sessions are using a single operating
+system-level network connection. This does not mean that the operating system
+terminates all sessions or network access; it only ends the inactive session
+and releases the resources associated with that session.
   "
   impact 0.5
-  tag "severity": "medium"
   tag "gtitle": "SRG-OS-000163-GPOS-00072"
   tag "gid": "V-72223"
-  tag "rid": "SV-86847r2_rule"
+  tag "rid": "SV-86847r3_rule"
   tag "stig_id": "RHEL-07-040160"
-  tag "cci": "CCI-001133"
-  tag "nist": ["SC-10", "Rev_4"]
-  tag "cci": "CCI-002361"
-  tag "nist": ["AC-12", "Rev_4"]
+  tag "cci": ["CCI-001133", "CCI-002361"]
+  tag "documentable": false
+  tag "nist": ["SC-10", "AC-12", "Rev_4"]
   tag "check": "Verify the operating system terminates all network connections
 associated with a communications session at the end of the session or based on
 inactivity.
 
 Check the value of the system inactivity timeout with the following command:
 
-# grep -i tmout /etc/bashrc
+# grep -i tmout /etc/bashrc /etc/profile.d/*
+
 TMOUT=600
 
-If \"TMOUT\" is not set to \"600\" or less in \"/etc/bashrc\", this is a finding."
-  tag "fix": "Configure the operating system to terminate all network connections
-associated with a communications session at the end of the session or after a period
-of inactivity.
+If \"TMOUT\" is not set to \"600\" or less in \"/etc/bashrc\" or in a script
+created to enforce session termination after inactivity, this is a finding."
+  tag "fix": "Configure the operating system to terminate all network
+connections associated with a communications session at the end of the session
+or after a period of inactivity.
 
-Add the following line to \"/etc/profile\" (or modify the line to have the required
-value):
+Add or update the following lines in \"/etc/profile\".
 
 TMOUT=600
+readonly TMOUT
+export TMOUT
+
+Or create a script to enforce the inactivity timeout (for example
+/etc/profile.d/tmout.sh) such as:
+
+#!/bin/bash
+
+TMOUT=600
+readonly TMOUT
+export TMOUT"
+  tag "fix_id": "F-78577r4_fix"
+
+  bashrc_file = parse_config_file('/etc/bashrc')
 
-The SSH service must be restarted for changes to take effect."
+  describe.one do
+    describe bashrc_file do
+      its('TMOUT') { should cmp <= SYSTEM_INACTIVITY_TIMEOUT }
+    end
 
-  describe parse_config_file('/etc/bashrc') do
-    its('TMOUT.to_i') { should cmp <= 600 }
+    profiled_files = command("find /etc/profile.d/*").stdout.split("\n")
```

Comment:
This is arbitrary. The only actual way to verify that this is happening properly is to log into the system as an actual user and see what the environment variable is set to.

This doesn't have to be changed right now, but it needs to be updated at some point.